pkg/cmd/build/build.go

```diff
@@ -0,0 +1,433 @@
+   limitations under the License.
+*/
+
+package build
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/images/archive"
+	"github.com/containerd/containerd/platforms"
+	dockerreference "github.com/containerd/containerd/reference/docker"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/buildkitutil"
+	"github.com/containerd/nerdctl/pkg/clientutil"
+	"github.com/containerd/nerdctl/pkg/platformutil"
+	"github.com/containerd/nerdctl/pkg/strutil"
+	"github.com/sirupsen/logrus"
+)
+
+func Build(ctx context.Context, options *types.BuildCommandOptions, stdin io.Reader, stdout, stderr io.Writer) error {
+	platform := options.Platform
+	buildctlBinary, buildctlArgs, needsLoading, metaFile, tags, cleanup, err := generateBuildctlArgs(ctx, stdin, options)
+	if err != nil {
+		return err
+	}
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	quiet := options.Quiet
+
+	logrus.Debugf("running %s %v", buildctlBinary, buildctlArgs)
+	buildctlCmd := exec.Command(buildctlBinary, buildctlArgs...)
+	buildctlCmd.Env = os.Environ()
+
+	var buildctlStdout io.Reader
+	if needsLoading {
+		buildctlStdout, err = buildctlCmd.StdoutPipe()
+		if err != nil {
+			return err
+		}
+	} else {
+		buildctlCmd.Stdout = stdout
+	}
+	if !quiet {
+		buildctlCmd.Stderr = stderr
+	}
+
+	if err := buildctlCmd.Start(); err != nil {
+		return err
+	}
+
+	if needsLoading {
+		platMC, err := platformutil.NewMatchComparer(false, platform)
+		if err != nil {
+			return err
+		}
+		if err = loadImage(ctx, buildctlStdout, options.GOptions.Namespace, options.GOptions.Address, options.GOptions.Snapshotter, stdout, platMC, quiet); err != nil {
+			return err
+		}
+	}
+
+	if err = buildctlCmd.Wait(); err != nil {
+		return err
+	}
+
+	iidFile := options.IidFile
+	if iidFile != "" {
+		id, err := getDigestFromMetaFile(metaFile)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(iidFile, []byte(id), 0600); err != nil {
+			return err
+		}
+	}
+
+	client, ctx, cancel, err := clientutil.NewClient(ctx, options.GOptions.Namespace, options.GOptions.Address)
+	if len(tags) > 1 {
+		logrus.Debug("Found more than 1 tag")
+		if err != nil {
+			return fmt.Errorf("unable to tag images: %s", err)
+		}
+		defer cancel()
+		imageService := client.ImageService()
+		image, err := imageService.Get(ctx, tags[0])
+		if err != nil {
+			return fmt.Errorf("unable to tag image: %s", err)
+		}
+		for _, targetRef := range tags[1:] {
+			image.Name = targetRef
+			if _, err := imageService.Create(ctx, image); err != nil {
+				// if already exists; skip.
+				if errors.Is(err, errdefs.ErrAlreadyExists) {
+					continue
+				}
+				return fmt.Errorf("unable to tag image: %s", err)
+			}
+		}
```

Comment:
can you check the whole line 103-125? It seems wrong to me.. I guess it might due to indent while refactor.

In the current format, nothing is done if `len(tags) == 0`. Should it be like this?:

```golang
	client, ctx, cancel, err := clientutil.NewClient(ctx, options.GOptions.Namespace, options.GOptions.Address)
	if err != nil {
		return fmt.Errorf("unable to tag images: %s", err)
	}
	defer cancel()
	if len(tags) > 1 {
		logrus.Debug("Found more than 1 tag")
	}
	imageService := client.ImageService()
	image, err := imageService.Get(ctx, tags[0])
	if err != nil {
		return fmt.Errorf("unable to tag image: %s", err)
	}
	for _, targetRef := range tags[1:] {
		image.Name = targetRef
		if _, err := imageService.Create(ctx, image); err != nil {
			// if already exists; skip.
			if errors.Is(err, errdefs.ErrAlreadyExists) {
				continue
			}
			return fmt.Errorf("unable to tag image: %s", err)
		}
	}
```